ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -905,6 +905,12 @@ private Offset generateReferenceIndexSegmentStartOffset(Offset recordOffset) {
       if (value.isFlagSet(IndexValue.Flags.Delete_Index)) {
         // delete record is always valid
         validEntries.add(new IndexEntry(key, value));
+        if (!isDeletedAt(key, deleteReferenceTimeMs) && !isExpiredAt(key, expiryReferenceTimeMs)
+            && value.getOriginalMessageOffset() != -1
```

Comment:
I was wondering about the same thing during implementation. It makes it slightly easier to read when checking it explicitly but yes the next condition encapsulates this check.